Ch setup database

BenoitIT opened the pull request.

Aimeana100 committed:
chore(Project-initialization) : Project setup

This is the main entry point of the project and Folder structure with best practice

- `Project setup`
- ``Initial dependencies installation``
-  `dontenv`
-  `express`
-  `body-parser`
- ``Folder structurring``
-  `src`
-   ```api``` : This folder contains all the API routes.
-   ```config``` : This folder contains configuration files for the application.
-   ```controllers``` : This folder contains the business logic of the application.
-   ```models``` : This folder contains the data models of the application.
-   ```middlewares``` : This folder contains all the middleware functions for the application.
-   ```utils``` : This folder contains all the utility functions for the application.
- docs: This folder contains all the documentation files for the application.
- public: This folder contains all the public files such as images, stylesheets, and client-side scripts.
- node_modules: This folder contains all the installed dependencies for the application.
- package.json: This file contains the metadata of the application and its dependencies.
- README.md: This file contains the documentation of the application.

Aimeana100 committed:
ft(default-endpoint): create  a default endpoint

This is to add default response with appropriate to the request format

- created a default api (root) in `src/api/root` directory
- added views forlder to save ```html``` files and created the default `index.html` and `404.htl`
- added middlewares for the Routes created in `src/index.js`

Aimeana100 committed:
ft(fix-Eslint and Prettier bugs):configured Eslint
 and Prettier

- added .eslintrc.js and .eslintignore files for configuring Eslint

Aimeana100 committed:
ch(add-README) : updated a README.md file

- updated a README.md file
- describe project flow and file structure

Aimeana100 committed:
ch(setup-API-docs) : setup API documentation

This is the the API documentation for the setup API, using swagger UI libraly

- install swagger UI libraly `npm install swagger-ui-express`
- create folder for docs at root Directory and basic libraly testing files.
-  `docs/index.js` : documentation running scripts
-  `docs/home.js` : docummented routes
-  `docs/schema.js` : schema documentation (structure ) to test routes based on.

Aimeana100 committed:
ch(setup-API-docs) : setup API documentation

    This is the the API documentation for the setup API, using swagger UI libraly

    - install swagger UI libraly `npm install swagger-ui-express`
    - create folder for docs at root Directory and basic libraly testing files.
    -  `docs/index.js` : documentation running scripts
    -  `docs/home.js` : docummented routes
    -  `docs/schema.js` : schema documentation (structure ) to test routes based on.
    - update README.dm file with the updates => Documentations

BenoitIT committed:
chore[postgres database setup]

adding sequelize package
adding sequelize-cli,pg and pg-hstore
initializing sequelize-cli
configure postgres with sequelize
create first migration which is user migration
run db migration
create function to add new user
create a new end point to add user to test database and app interconnectivity
## how to test this
run npm init
open postman
run localhost:4000/new-user
insert email,firstname,lastname,password in Json format
send request
wait for the response which states that the user is created sucessessfully
[finishing testing postgres and app interconnectivity]

mukunzidd reviewed (changes requested):
@BenoitIT please a clear detailed description of your PR and add steps to configure Postgres locally in the Readme.md

BenoitIT committed:
chore[database setup]

##what does this commit have

-function to display connection message
-message to confirm connectivity
##how to test this codes
-run a server
-using npm run dev
[finishing commit]

BenoitIT commented:
thanks done!

On Tue, Feb 28, 2023 at 7:35 AM mukunzidd ***@***.***> wrote:

> ***@***.**** requested changes on this pull request.
>
> @BenoitIT <https://github.com/BenoitIT> please a clear detailed
> description of your PR and add steps to configure Postgres locally in the
> Readme.md
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/atlp-rwanda/ecommerce-app-legends-bn/pull/2#pullrequestreview-1316828622>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AYULN6Z642G2IEHLO5REYLDWZWFAHANCNFSM6AAAAAAVJ7L76E>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
